bin/templates/cordova/Api.js

```diff
@@ -81,10 +83,13 @@ function Api(platform, platformRootDir, events) {
     // XXX Override some locations for Android Studio projects
     if(AndroidStudio.isAndroidStudioProject(self.root) === true) {
       selfEvents.emit('log', 'Android Studio project detected');
+      this.builder='studio';
       this.android_studio = true;
       this.locations.configXml = path.join(self.root, 'app/src/main/res/xml/config.xml');
-      this.locations.strings = path.join(self.root, 'app/src/main/res/xml/strings.xml');
+      this.locations.strings = path.join(self.root, 'app/src/main/res/values/strings.xml');
       this.locations.manifest = path.join(self.root, 'app/src/main/AndroidManifest.xml');
+      //We could have Java Source, we could have other languages
+      this.locations.javaSrc = path.join(self.root, 'app/src/main/java/');
```

Comment:
Same sort of deal, we could push the responsibility for figuring out the locations to the builders, rather than have the Api function worry about each possible implementation of a project structure.